Improve API docs (#341)
ballerina-platform/module-ballerinax-java.jdbc: docs/spec/spec.md

_Owners_: @daneshk @niveathika  
_Reviewers_: @daneshk  
_Created_: 2022/01/14   
_Updated_: 2022/01/25  
_Issue_: [#2290](https://github.com/ballerina-platform/ballerina-standard-library/issues/2290)

# Introduction

> This is the specification for the JDBC standard library, which provides the functionality that is required to access and 

Reviewer: ```suggestion
This is the specification for the JDBC standard library, which provides the functionality that is required to access and 
```
```suggestion
This is the specification for the JDBC standard library, which provides the functionality that is required to access and 
```
This sentence is too long. Consider breaking it into multiple  sentences.

Author: Fixed in fc94727

2. Executes the query, which is expected to return at most one row of the result.
3. Executes the SQL query. Only the metadata of the execution is returned.
4. Executes the SQL query with multiple sets of parameters in a batch. Only the metadata of the execution is returned.
5. Executes an SQL query, which calls a stored procedure. This can either return results or nil.

> All the above operations make use of `sql:ParameterizedQuery` object, backtick surrounded string template to pass

Reviewer: ```suggestion
All the above operations make use of the sql:ParameterizedQuery` object and the string template surrounded by backticks to pass
```

Author: Fixed in fc94727

**Configurations available for initializing the JDBC client:**
* Connection properties:
   ```ballerina
   # Provides an additional set of configurations related to a database connection.
   #
   # + datasourceName - The driver class name to be used to get the connection

>    # + properties - The database properties which should be applied when getting the connection

Reviewer: ```suggestion
   # + properties - The database properties which should be applied when getting the connection
```
```suggestion
   # + properties - The database properties, which should be applied when getting the connection
```

Author: Fixed in fc94727

## 2.1. Connection Pool Handling

> Connection pool handling is generic and implemented through `sql` module. For more information, see the 

Author: Fixed in fc94727

operation. This will close the corresponding connection pool if it is not shared by any other database clients.

   ```ballerina
   # Closes the JDBC client and shuts down the connection pool.
   #
   # + return - Possible error when closing the client
   public isolated function close() returns Error?;
   ```

# 3. Queries and Values

All the generic `sql` queries and values are supported. For more information, see the 

> [SQL Specification](https://github.com/ballerina-platform/module-ballerina-sql/blob/master/docs/spec/spec.md#3-queries-and-values)

Reviewer: ```suggestion
[SQL Specification](https://github.com/ballerina-platform/module-ballerina-sql/blob/master/docs/spec/spec.md#3-queries-and-values).
```

Author: Fixed in fc94727
